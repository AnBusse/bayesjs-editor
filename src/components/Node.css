.node {
  background-color: #ff9;
  border: solid 2px #111;
  border-radius: 50%;
  width: 100px;
  height: 100px;
  line-height: 100px;
  text-align: center;
  font-family: 'Segoe UI';
  font-size: 18px;
}
