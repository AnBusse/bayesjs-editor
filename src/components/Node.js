import React, { PropTypes } from 'react';
import { StyleSheet, css } from 'aphrodite';

const styles = StyleSheet.create({
  node: {
    cursor: 'move',
  },
});

const renderState = (state, index) => (
  <text key={state} x="5" y={37 + (18 * index)}>
    <tspan>{state}</tspan>
    <tspan x="155" textAnchor="end">99.9</tspan>
  </text>
);

const Node = props => (
  <g
    className={css(styles.node)}
    onMouseDown={props.onMouseDown}
    transform={`translate(${props.x} ${props.y})`}
  >
    <rect
      height={20 + (20 * props.states.length)}
      width="160"
      fill="#ff8"
      stroke="#333"
      ref={props.rectRef}
    />

    <text x="5" y="15">{props.id}</text>
    <path d="M0,20 h160" stroke="#333" />

    {props.states.map(renderState)}
  </g>
);

Node.propTypes = {
  id: PropTypes.string.isRequired,
  states: PropTypes.arrayOf(PropTypes.string).isRequired,
  rectRef: PropTypes.func,
  onMouseDown: PropTypes.func,
  x: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  y: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
};

export default Node;
