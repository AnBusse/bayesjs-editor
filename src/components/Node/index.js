import { nodePosition, sizePropTypes, statesPropTypes } from 'models';

import GenericNode from 'components/GenericNode';
import NodeConnections from 'components/NodeConnections';
import NodeStates from 'components/NodeStates';
import PropTypes from 'prop-types';
import React from 'react';

const Node = props => (
  <GenericNode {...props}>
    <NodeStates {...props} />
    <NodeConnections {...props} />
  </GenericNode>
);

Node.defaultProps = {
  belief: null,
  description: null,
};

Node.propTypes = {
  id: PropTypes.string.isRequired,
  states: statesPropTypes.isRequired,
  results: PropTypes.objectOf(PropTypes.number).isRequired,
  isSelected: PropTypes.bool.isRequired,
  belief: PropTypes.string,
  onMouseDown: PropTypes.func.isRequired,
  onStateDoubleClick: PropTypes.func.isRequired,
  position: nodePosition.isRequired,
  size: sizePropTypes.isRequired,
  description: PropTypes.string,
};

export default Node;
