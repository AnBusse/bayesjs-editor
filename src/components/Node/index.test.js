import React from 'react';
import { shallow } from 'enzyme';
import Node from './index';

const defaultProps = {
  id: 'id',
  states: ['S1', 'S2'],
  isSelected: false,
  onMouseDown: jest.fn(),
  onStateDoubleClick: jest.fn(),
  position: { x: 0, y: 0 },
  size: { width: 160, height: 100 },
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<Node {...compProps} />);
};

describe('Node Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
