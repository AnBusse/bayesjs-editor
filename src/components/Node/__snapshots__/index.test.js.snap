// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node Component matches snapshot 1`] = `
<GenericNode
  belief={null}
  description={null}
  id="id"
  isSelected={false}
  onMouseDown={[Function]}
  onStateDoubleClick={[Function]}
  position={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  size={
    Object {
      "heigth": 100,
      "width": 160,
    }
  }
  states={
    Array [
      "S1",
      "S2",
    ]
  }
>
  <NodeStates
    belief={null}
    description={null}
    id="id"
    isSelected={false}
    onMouseDown={[Function]}
    onStateDoubleClick={[Function]}
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    size={
      Object {
        "heigth": 100,
        "width": 160,
      }
    }
    states={
      Array [
        "S1",
        "S2",
      ]
    }
  />
  <NodeConnections
    belief={null}
    description={null}
    id="id"
    isSelected={false}
    onMouseDown={[Function]}
    onStateDoubleClick={[Function]}
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    size={
      Object {
        "heigth": 100,
        "width": 160,
      }
    }
    states={
      Array [
        "S1",
        "S2",
      ]
    }
  />
</GenericNode>
`;
